### PR TITLE
[#13047][Docs][DocDB] Documentation for tablet split gflags and usage in 2.14

### PR DESCRIPTION
**Pull request by jmeehan16**
Documentation changes for tablet splitting in 2.14. Includes descriptions for each of the relevant gflags and some updates to tablet splitting page with recommendations for best configurations for performance.

### CODE REVIEW

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 1]
```diff
@@ -282,8 +312,8 @@ In the following example, a three-node cluster is created and uses a YCSB worklo
 The following known limitations are planned to be resolved in upcoming releases:
 
 * Colocated tables cannot be split. For details, see [#4463](https://github.com/yugabyte/yugabyte-db/issues/4463)
-* Tablet splitting should be disabled during an index backfill. For details, see [#6704](https://github.com/yugabyte/yugabyte-db/issues/6704)
-* Cross cluster replication currently does not work with tablet splitting. For details, see [#5373](https://github.com/yugabyte/yugabyte-db/issues/5373)
+* By default, tablet splitting is disabled by default for tables with cross cluster replication. It can be enabled by setting both [`enable_tablet_split_of_xcluster_replicated_tables`](../../../reference/configuration/yb-master/#enable-tablet-split-of-xcluster-replicated-tables) and [`enable_tablet_split_of_xcluster_bootstrapping_tables`](../../../reference/configuration/yb-master/#enable-tablet-split-of-xcluster-bootstrapping-tables) to `true`.
```
Comment (jmeehan16) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:315]: "by default" twice here

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 2]
```diff
@@ -211,15 +211,45 @@ For details on the architecture design, see [Automatic re-sharding of data with
...
 
-* In the high phase, each node has fewer than [`tablet_split_high_phase_shard_count_per_node`](../../../reference/configuration/yb-master/#tablet-split-high-phase-shard-count-per-node) shards. In this phase, YugabyteDB splits tablets larger than [`tablet_split_high_phase_size_threshold_bytes`](../../../reference/configuration/yb-master/#tablet-split-high-phase-size-threshold-bytes).
+#### Low Phase
 
-* Once the shard count exceeds the high phase count, YugabyteDB splits tablets larger than [`tablet_force_split_threshold_bytes`](../../../reference/configuration/yb-master/#tablet-force-split-threshold-bytes).
+In the low phase, each node has fewer than [`tablet_split_low_phase_shard_count_per_node`](../../../reference/configuration/yb-master/#tablet-split-low-phase-shard-count-per-node) shards. In this phase, YugabyteDB splits tablets larger than [`tablet_split_low_phase_size_threshold_bytes`](../../../reference/configuration/yb-master/#tablet-split-low-phase-size-threshold-bytes).
+
+By default, a given table will be considered to be in the low phase until there are 8 tablets per node. In this phase, a tablet will be split once it reaches 512 MB in size.
```
Comment (SrivastavaAnubhav) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:222]: I think it's clear enough to say that the default values of `tablet_split_low_phase_shard_count_per_node` is 8, and the default value of `tablet_split_low_phase_size_threshold_bytes` is `512 MB`.

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 3]
```diff
@@ -211,15 +211,45 @@ For details on the architecture design, see [Automatic re-sharding of data with
...
+
+#### High Phase
+
+In the high phase, each node has fewer than [`tablet_split_high_phase_shard_count_per_node`](../../../reference/configuration/yb-master/#tablet-split-high-phase-shard-count-per-node) shards. In this phase, YugabyteDB splits tablets larger than [`tablet_split_high_phase_size_threshold_bytes`](../../../reference/configuration/yb-master/#tablet-split-high-phase-size-threshold-bytes).
+
+By default, a table will be considered to be in the high phase until there are 24 tablets per node. In this phase, a tablet will be split once it reaches 10 GB in size.
```
Comment (SrivastavaAnubhav) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:228]: Same comment as above (we should be clear that 10 GB is just the default value of `tablet_split_high_phase_size_threshold_bytes`).

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 4]
```diff
@@ -211,15 +211,45 @@ For details on the architecture design, see [Automatic re-sharding of data with
...
+
+#### Final Phase
+
+Once the shard count exceeds the high phase count, YugabyteDB splits tablets larger than [`tablet_force_split_threshold_bytes`](../../../reference/configuration/yb-master/#tablet-force-split-threshold-bytes). This will continue until we reach the [`tablet_split_limit_per_table`](../../../reference/configuration/yb-master/#tablet-split-limit-per-table) tablets per table limit (if set to 0, there will be no limit).
+
+By default, a given table will be considered in this final phase once there are more than 24 tablets per node (determined by `tablet_split_high_phase_shard_count_per_node`). Once in this phase, a tablet will be split once it reaches 100 GB in size. Splits can continue in this phase until there are 256 tablets for this table.
```
Comment (SrivastavaAnubhav) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:234]: "considered **to be** in"?
"once there are more than 24" -> "once there are more than `tablet_split_high_phase_shard_count_per_node`"

Same comment about being clear that the numbers we're quoting are just defaults controlled by the flags.

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 5]
```diff
@@ -211,15 +211,45 @@ For details on the architecture design, see [Automatic re-sharding of data with
...
+
+### Post-split compactions
+
+Once a split has been executed on a tablet, the two resulting tablets will require a full compaction in order to remove unnecessary data from each. These post-split compactions can require fairly significant CPU and disk resources, and impact workload performance as a result. To limit this impact, the number of simultaneous tablet splits allowed is conservative by default.
```
Comment (SrivastavaAnubhav) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:238]: Can shorten "can require fairly significant CPU and disk resources, and impact workload performance as a result". E.g to "can significantly increase CPU and disk usage and thus impact performance".

**Review by SrivastavaAnubhav — APPROVED** [review]
Looks good, just minor comments.

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md (hunk 1 above)
Comment (hulien22) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:315]: `enable_tablet_split_of_xcluster_bootstrapping_tables` should be kept as false until https://github.com/yugabyte/yugabyte-db/issues/13170 gets landed, as xcluster bootstrapping + tablet splitting is currently unsupported

**Review by hulien22 — CHANGES_REQUESTED** [review]

**Comment by polarweasel** [comment]
You guys all need to use the "Suggest change" button :)

File: docs/content/preview/reference/configuration/yb-master.md [hunk 6]
```diff
@@ -471,30 +477,122 @@ On a per-table basis, the [`CREATE TABLE ...SPLIT INTO`](../../../api/ysql/the-s
...
+
+If `enable_automatic_tablet_splitting` is `true`, this value will be overridden and tables will begin with 1 tablet per node.
+
+{{< /note >}}
+
+### Tablet splitting flags
+
 ##### --enable_automatic_tablet_splitting
 
 Enables YugabyteDB to [automatically split tablets](../../../architecture/docdb-sharding/tablet-splitting/#automatic-tablet-splitting) while online, based on the specified tablet threshold sizes configured below.
```
[the remark this answers is not in the document]
Comment (jmeehan16) [reply docs/content/preview/reference/configuration/yb-master.md:490]: Good question, haha. I think we can just remove those words.

**Comment by polarweasel** [comment]
@jmeehan16 `docs/content/preview` is v2.15, and `docs/content/stable` is v2.14. Did you need these changes in 2.15 (preview), 2.14 (stable), or both?

**Comment by jmeehan16** [comment]
> docs/content/preview is v2.15, and docs/content/stable is v2.14. Did you need these changes in 2.15 (preview), 2.14 (stable), or both?
@polarweasel - Good point. This needs to be applied to both 2.15 (preview) and 2.14 (stable).

**Review by polarweasel — APPROVED** [review]
I've applied the changes to stable now. LGTM!

**Pull request merged**